### ISSUE TITLE
Uninstrument BarrierRectangle in favor of display.addInputListener

### ISSUE THREAD

**Issue by zepumph** [phetsims/joist#499]
This is a new pattern that seems better for phetio because then we can uninstrument BarrierRectangle, simplifying the phetio api for monitoring dialog visibility.

In `Sim.js` something like:
```
      this.display.addInputListener( {
        down: function() {
          // self.startEvent( 'user', 'fired' );
          assert && assert( self.modalNodeStack.length > 0, 'There must be a Node in the stack to hide.' );
          self.modalNodeStack.get( self.modalNodeStack.length - 1 ).hide();
          // self.endEvent();
        }
      } );
```

copied from `BarrierRectangle`'s  button listener.

**Comment by zepumph** [comment]
This should be explored before FL "thoughtful" release.

@samreid's philosophy "good luck, hopefully it works out, but maybe not."

**Comment by zepumph** [comment]
@samreid I'm having a bit of trouble thinking about how to go about doing this. I feel like the pattern applies more to each place where the modalNodeStack is used, rather than with `BarrierRectangle`. I looked and `ComboBox` and think that pattern would parallel by having `Dialog` handle this, on `Dialog.show()` it would add a listener to the display, and it could remove it (and itself) on `hide()`.

If I'm not understanding this fully, I would love to have a converstation to get on the same page.

**Comment by samreid** [comment]
I think you described the right way to implement it, but the waters are muddy because we do want to dim the background, so the BarrierRectangle or something like it may be warranted anyways.

**Comment by zepumph** [comment]
I would like to dive into this together, marking as high priority with the phet-io meeting tag.

**Comment by zepumph** [comment]
Assigning so this doesn't fall off the radar.

**Comment by samreid** [comment]
In discussion today, we decided the BarrierRectangle is the current implementation approach in Sim.js and unlikely to change in the near future.  Having the IO API match the Sim.js implementation approach is appropriate for now.  If we change Sim.js to use display events in the future (with some other way of dimming the background), then it will be appropriate to change the phet-io API as well.  In https://github.com/phetsims/axon/issues/180 @zepumph is going to look at converting `events: ['fired']` to use Emitters, and that will cover BarrierRectangleIO as well.  Closing.
